documentation/interrupt_handling.md

```diff
@@ -28,22 +84,20 @@ Two types of interrupt are defined in optee_os:
 For ARM GICv2 mode, native interrupt is sent as FIQ and foreign interrupt is
 sent as IRQ.
 
-Since IRQs are received using the state vector the actual vector used
-depends on the current state of the CPU. If the NS (non-secure) bit in SCR
-(Secure Control Register) is set then either HVBAR or VBAR (non-secure) is
-used when receiving the IRQ, if the NS bit in SCR is cleared the secure
-VBAR is used instead. This has the consequence that secure world can
-receive IRQ that are supposed to be handled by normal world. When secure
-world receives an IRQ it has to be forwarded to normal world for
-processing.
-
 For ARM GICv3 mode, foreign interrupt is sent as FIQ which could be handled
 by either secure world (aarch32 Monitor mode or aarch64 EL3) or normal world.
 ARM GICv3 mode can be enabled by setting `CFG_ARM_GICV3=y`.
 
-The following descriptions take ARM GICv2 mode as the example.
+For clarity, this document mainly choose the GICv2 convention and refers the
+IRQ as optee_os foreign interrupts, and FIQ as optee_os native interrupts.
+
+Native interrupts must be securely routed to optee_os. Foreign interrupts,
+when trapped during secure world execution might need to be efficiently routed
+to the normal world.
 
-# The monitor
+# 2. Normal World Invokes the optee_os Through a SMC
+
+## 2.1. Entering the Monitor
```

Comment:
Entering the secure monitor

Reply:
Would you prefer i replace all occurrences of "Monitor" with "secure monitor" ?

Reply:
No, only where it feels natural to use the full name.